src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstringTests.Fixer.cs

```diff
@@ -0,0 +1,215 @@
+        Dim sb = New StringBuilder()
+        sb.Append(text, 0, 6)
+        Return sb.ToString()
+    End Function
+End Class
+";
+            VerifyBasicFix(Code, FixedCode);
+        }
+
+        [Fact]
+        public void FixesExample2FromTicket()
+        {
+            const string Code = @"
+using System.Text;
+
+public class C
+{
+    public string Append(string text)
+    {
+        var sb = new StringBuilder();
+        sb.Append(text.Substring(2));
+        return sb.ToString();
+    }
+}";
+            const string FixedCode = @"
+using System.Text;
+
+public class C
+{
+    public string Append(string text)
+    {
+        var sb = new StringBuilder();
+        sb.Append(text, 2, text.Length - 2);
+        return sb.ToString();
+    }
+}";
+            VerifyCSharpFix(Code, FixedCode);
+        }
+
+        [Fact]
+        public void FixesExample2FromTicketBasic()
+        {
+            const string Code = @"
+Imports System.Text
+
+Public Class C
+    Public Function Append(ByVal text As String) As String
+        Dim sb = New StringBuilder()
+        sb.Append(text.Substring(2))
+        Return sb.ToString()
+    End Function
+End Class
+";
+            const string FixedCode = @"
+Imports System.Text
+
+Public Class C
+    Public Function Append(ByVal text As String) As String
+        Dim sb = New StringBuilder()
+        sb.Append(text, 2, text.Length - 2)
+        Return sb.ToString()
+    End Function
+End Class
+";
+            VerifyBasicFix(Code, FixedCode);
+        }
+
+        [Fact(Skip = "not working yet")]
+        public void FixesAllExamplesFromTicket()
+        {
+
+            const string Code = @"
+using System.Text;
+
+public class C
+{
+    public string Append (string text)
+    {
+        var sb = new StringBuilder();
+        sb.Append(text.Substring (0, 6));
+        sb.Append(text.Substring (2));
+        return sb.ToString ();
+    }
+}";
+            const string FixedCode = @"
+using System.Text;
+
+public class C
+{
+    public string Append(string text)
+    {
+        var sb = new StringBuilder();
+        sb.Append(text, 0, 6);
+        sb.Append(text, 2, text.Length - 2);
+        return sb.ToString ();
+    }
+}";
+            VerifyCSharpFixAll(Code, FixedCode);
+        }
+
+        [Fact(Skip = "not working yet")]
+        public void FixesAllExamplesFromTicketBasic()
+        {
+
+            const string Code = @"
+Imports System.Text
+
+Public Class C
+    Public Function Append(ByVal text As String) As String
+        Dim sb = New StringBuilder()
+        sb.Append(text.Substring(0, 6))
+        sb.Append(text.Substring(2))
+        Return sb.ToString()
+    End Function
+End Class
+";
+            const string FixedCode = @"
+Imports System.Text
+
+Public Class C
+    Public Function Append(ByVal text As String) As String
+        Dim sb = New StringBuilder()
+        sb.Append(text, 0, 6)
+        sb.Append(text, 2, text.Length - 2)
+        Return sb.ToString()
```

Comment:
need tests (for both C# and VB) where "sb.Append(...)" is the argument to something.  it will validate that the getInnermostForTie code is working properly.

Reply:
Added ReplaceWhenSbAppendIsAParameterToValidateGetInnermostForTie and ReplaceWhenSbAppendIsAParameterToValidateGetInnermostForTieBasic for this case.